python/GangaDirac/Lib/Backends/DiracBase.py

```diff
@@ -495,20 +497,25 @@ def _getStateTime(job, status):
             if not job.subjobs and status in b_list:
                 for childstatus in b_list:
                     if job.backend.id:
-                        logger.debug("Accessing getStateTime() in diracAPI")
-                        dirac_cmd = "getStateTime(%d,\'%s\')" % (job.backend.id, childstatus)
-                        be_statetime = execute(dirac_cmd)
+               #         logger.debug("Accessing getStateTime() in diracAPI")
+                        #be_statetime = execute("getStateTime(%d,\'%s\')" % (job.backend.id, childstatus))
                         if childstatus in backend_final:
+                            be_statetime = execute("getStateTime(%d,\'%s\')" % (job.backend.id, childstatus))
                             job.time.timestamps["backend_final"] = be_statetime
-                            logger.debug("Wrote 'backend_final' to timestamps.")
+                            break
+               #             logger.debug("Wrote 'backend_final' to timestamps.")
                         else:
-                            job.time.timestamps["backend_" + childstatus] = be_statetime
-                            logger.debug("Wrote 'backend_%s' to timestamps.", childstatus)
+                            time_str = "backend_" + childstatus
+                            if time_str not in job.time.timestamps:
+                                be_statetime = execute("getStateTime(%d,\'%s\')" % (job.backend.id, childstatus))
+                                job.time.timestamps["backend_" + childstatus] = be_statetime
+               #             logger.debug("Wrote 'backend_%s' to timestamps.", childstatus)
                     if childstatus == status:
                         break
-            logger.debug("_getStateTime(job with id: %d, '%s') called.", job.id, job.status)
+            #logger.debug("_getStateTime(job with id: %d, '%s') called.", job.id, job.status)
         else:
-            logger.debug("Status changed from '%s' to '%s'. No new timestamp was written", job.status, status)
+            pass
+            #logger.debug("Status changed from '%s' to '%s'. No new timestamp was written", job.status, status)
```

Comment:
I'm yet to confirm if the logger here was causing a performance problem. I suspect not so the logging will probably go back in before merge